File: src/main/java/io/github/svegon/capi/event/render/tooltip/CommandBlockTooltipCallback.java
package io.github.svegon.capi.event.render.tooltip;

import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;
import net.minecraft.block.CommandBlock;
import net.minecraft.client.item.TooltipContext;
import net.minecraft.item.ItemStack;
import net.minecraft.text.Text;
import net.minecraft.world.BlockView;
import org.jetbrains.annotations.Nullable;

import java.util.List;

@FunctionalInterface
public interface CommandBlockTooltipCallback {
    Event<CommandBlockTooltipCallback> EVENT = EventFactory.createArrayBacked(CommandBlockTooltipCallback.class,
            (block,stack, world, tooltip, options) -> {}, listeners -> (block,stack, world, tooltip, context) -> {
        for (CommandBlockTooltipCallback listener : listeners) {
            listener.appendCommandBlockTooltip(block, stack, world, tooltip, context);
        }
    });

    void appendCommandBlockTooltip(CommandBlock block, ItemStack stack, @Nullable BlockView world,
                                   List<Text> tooltip, TooltipContext options);
}

File: src/main/kotlin/io/github/svegon/mclientapi/event/render/tooltip/CommandBlockTooltipCallback.kt
package io.github.svegon.capi.event.render.tooltip

import net.fabricmc.fabric.api.event.Event
import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.client.item.TooltipContext
import net.minecraft.text.Text
import net.minecraft.world.BlockView

fun interface CommandBlockTooltipCallback {
    fun appendCommandBlockTooltip(
        block: CommandBlock?, stack: ItemStack?, world: BlockView?,
        tooltip: List<Text?>?, options: TooltipContext?
    )

    companion object {
        @JvmField
        val EVENT: Event<CommandBlockTooltipCallback> = EventFactory.createArrayBacked<CommandBlockTooltipCallback>(
            CommandBlockTooltipCallback::class.java,
            CommandBlockTooltipCallback { block: CommandBlock?, stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, options: TooltipContext? -> }) { listeners: Array<CommandBlockTooltipCallback> ->
            CommandBlockTooltipCallback { block: CommandBlock?, stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, context: TooltipContext? ->
                for (listener in listeners) {
                    listener.appendCommandBlockTooltip(block, stack, world, tooltip, context)
                }
            }
        }
    }
}
